python/ray/_private/client_mode_hook.py

```diff
@@ -6,6 +6,7 @@
 RAY_CLIENT_MODE_ATTR = "__ray_client_mode_key__"
 
 client_mode_enabled = os.environ.get("RAY_CLIENT_MODE", "0") == "1"
+os.environ.update({"RAY_CLIENT_MODE": "0"})
```

Comment:
The timeout in client mode is caused by this one.
I'm not sure why it doesn't cause any problems before. :( 

When testing in client mode, this env will be passed to raylet too where default_worker.py will be constructed. And because this variable is set to be true which will make the default_worker run in client mode, so it'll complain something like, please connect first.
@ericl do you have any idea about this?

Reply:
Why do we need to change this in this PR?

Reply:
Ok, to make it clear:

1) e2e test running with RAY_CLIENT_MODE=1
2) it'll run raylet inside the process which will have RAY_CLIENT_MODE=1 too
3) raylet will spawn worker which is with RAY_CLIENT_MODE=1
4) the worker spawned by raylet will treat itself as a client, which will lead to error like: please connect first

It looks like a bug for me, but somehow never triggered before.